src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1563,6 +1563,7 @@
 
   <Node Name="BoundDeclarationPattern" Base="BoundPattern">
     <Field Name="LocalSymbol" Type="LocalSymbol" Null="disallow"/>
```

Comment:
This represents a declaration in source, and therefore is required to be present by the design invariants of the bound nodes.

---
In reply to: [77385499](https://github.com/dotnet/roslyn/pull/13535#discussion_r77385499) [](ancestors = 77385499)